shot/src/main/scala/com/karumi/shot/ShotPlugin.scala

```diff
@@ -72,4 +75,79 @@ class ShotPlugin extends Plugin[Project] {
     project.getExtensions.add(name, new ShotExtension())
   }
 
+  private def isLegacyGradleVersion(gradleVersion: String): Boolean = {
+    val versionNumbers = gradleVersion.split('.')
+    var major = 0
+    var minor = 0
+
+    if (versionNumbers.length > 0) {
+      major = versionNumbers(0).toInt
+    }
+    if (versionNumbers.length > 1) {
+      minor = versionNumbers(1).toInt
+    }
+
+    (major, minor) match {
+      case (major, _) if major > GRADLE_MIN_MAJOR => false
+      case (major, minor)
+          if major == GRADLE_MIN_MAJOR && minor >= GRADLE_MIN_MINOR =>
+        false
+      case _ => true
+    }
+
+  }
+
+  private def androidDependencyMode(project: Project): String = {
+    val connection = GradleConnector.newConnector
+      .forProjectDirectory(project.getProjectDir)
+      .connect
+
+    try {
+      val gradleVersion = connection
+        .getModel(classOf[BuildEnvironment])
+        .getGradle
+        .getGradleVersion
+
+      if (isLegacyGradleVersion(gradleVersion)) {
+        return Config.androidDependencyModeLegacy
+      }
+
+    } finally connection.close()
+
+    Config.androidDependencyMode
+  }
+
+  private def addAndroidTestDependency(project: Project): Unit = {
+
+    project.getGradle.addListener(new DependencyResolutionListener() {
+
+      override def beforeResolve(
+          resolvableDependencies: ResolvableDependencies): Unit = {
+        var facebookDependencyHasBeenAdded = false
+
+        project.getConfigurations.forEach(config => {
+          facebookDependencyHasBeenAdded |= config.getAllDependencies
+            .toArray(new Array[Dependency](0))
+            .exists(dependency =>
+              Config.androidDependencyGroup == dependency.getGroup
+                && Config.androidDependencyName == dependency.getName)
+        })
+
+        if (!facebookDependencyHasBeenAdded) {
+          val dependencyMode = androidDependencyMode(project)
+          val dependencyName = Config.androidDependency
+          val dependenciesHandler = project.getDependencies
+
+          val dependencyToAdd = dependenciesHandler.create(dependencyName)
+          Option(project.getPlugins.findPlugin(Config.androidPluginName))
+            .map(_ =>
+              project.getDependencies.add(dependencyMode, dependencyToAdd))
+          project.getGradle.removeListener(this)
+        }
+      }
```

Comment:
Nice research @flipper83 I had no idea we had to analyze the complete dependency tree. I thought it was going to be easier 😃  Thanks

Reply:
Update. Thank you @Serchinastico and @pedrovgs for the support :)